src/tink/core/Error.hx

```diff
@@ -42,11 +43,15 @@ class TypedError<T> {
 	public var code(default, null):ErrorCode;
 	public var data(default, null):T;
 	public var pos(default, null):Null<Pos>;
+	public var callStack(default, null):Array<StackItem>;
+	public var exceptionStack(default, null):Array<StackItem>;
 	
 	public function new(?code:ErrorCode = InternalError, message, ?pos) {
 		this.code = code;
 		this.message = message;
 		this.pos = pos;
+		this.callStack = try CallStack.callStack() catch(e:Dynamic) [];
+		this.exceptionStack = try CallStack.exceptionStack() catch(e:Dynamic) [];

```

Comment:
IIRC on some platform Haxe captures the call stack by throwing an error and capturing it right away. That probably resets the exception stack? Probably we should reverse these two lines of code to capture the exception stack first, before it is changed by the `callStack()` call.